src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -210,15 +224,15 @@ public boolean isFinal() {
         if (stateData.opt.arriveBy) {
             // Check that we are not renting a bike at the destination
             // Also check that a bike was rented if bikeRental is specified
-            bikeRentingOk = !isBikeRenting() && (!stateData.opt.bikeRental || hasUsedRentedBike());
+            bikeRentingOk = !stateData.opt.bikeRental || (bikeRentalNotStarted() || bikeRentalIsFinished());
```

Comment:
This check doesn't seem to be correct. The `bikeRentalIsFinished` method will return true if the current state is in `RENTING_FLOATING`. However, in an arriveBy context, this means that a user is in possession of a floating bike that they have yet to pickup from an actual station (docked or floating). Therefore, the check here should be only if the bike rental state is `BEFORE_RENTING` or `HAVE_RENTED`.